modules/pam_xauth/pam_xauth.c

```diff
@@ -442,7 +444,16 @@ pam_sm_open_session (pam_handle_t *pamh, int flags UNUSED,
 		retval = PAM_SESSION_ERR;
 		goto cleanup;
 	}
-	rpwd = pam_modutil_getpwuid(pamh, getuid());
+
+	login_name = pam_modutil_getlogin(pamh);
+	if (login_name == NULL) {
+		login_name = "";
+	}
+	if (*login_name)
+		rpwd = pam_modutil_getpwnam(pamh, login_name);
+	else
+		rpwd = pam_modutil_getpwuid(pamh, getuid());
```

Comment:
This essentially replaces `pam_modutil_getpwuid(pamh, getuid())` with `pam_modutil_getpwnam(pamh, pam_modutil_getlogin(pamh))` without any mentioning in the commit message.  What it has to do with NFS and the cookiefile?